apps/prairielearn/src/ee/webhooks/stripe/index.ts

```diff
@@ -0,0 +1,101 @@
+import express = require('express');
+import asyncHandler = require('express-async-handler');
+import Stripe from 'stripe';
+import error = require('@prairielearn/error');
+
+import { config } from '../../../lib/config';
+import { getStripeClient } from '../../lib/billing/stripe';
+import {
+  getStripeCheckoutSessionBySessionId,
+  markStripeCheckoutSessionCompleted,
+} from '../../models/stripe-checkout-sessions';
+import { runInTransactionAsync } from '@prairielearn/postgres';
+import { ensurePlanGrant } from '../../models/plan-grants';
+
+const router = express.Router({ mergeParams: true });
+
+function constructEvent(req: express.Request) {
+  if (!config.stripeWebhookSigningSecret) {
+    throw new Error('Stripe is not configured.');
+  }
+
+  const stripe = getStripeClient();
+  try {
+    return stripe.webhooks.constructEvent(
+      req.body,
+      req.headers['stripe-signature'] as string,
+      config.stripeWebhookSigningSecret,
+    );
+  } catch (err) {
+    throw error.make(400, `Webhook error: ${err.message}`);
+  }
+}
+
+async function handleSessionUpdate(session: Stripe.Checkout.Session) {
```

Comment:
Should we update our local view of the session as stored in the `stripe_checkout_sessions` table here? I'm inclined to say yes?

Reply:
We store a big JSON blob of the Checkout Session object in the `stripe_checkout_sessions` table (the `data` column). My note here was about updating that column here.

Reply:
I added code to persist the updated `checkout.session` object in 3ff6993.